SWLOR.Game.Server/Feature/ChatCommandDefinition/DMChatCommand.cs

```diff
@@ -406,10 +406,64 @@ private void SetLocalVariable()
 
                     SendMessageToPC(user, "Local string set: " + variableName + " = " + value);
                 });
+
+            _builder.Create("tptag")
+                .Description("Sets a local tag on a target Teleport Object placeable.")
+                .Permissions(AuthorizationLevel.DM, AuthorizationLevel.Admin)
+                .RequiresTarget()
+                .Validate((user, args) =>
+                {
+                    if (args.Length <= 0)
+                    {
+                        return "Missing arguments. Format should be: /TPTag <VALUE>. Example: /TPTag DUNGEON_ENTRANCE";
+                    }
+
+                    return string.Empty;
+                })
+                .Action((user, target, location, args) =>
+                {
+                    if (GetResRef(target) != "tele_obj")
+                    {
+                        SendMessageToPC(user, "This command can only be used on the Teleport Object placeable.");
+                    }
+                    else
+                    {
+                        SetTag(target, args[0]);
+
+                        SendMessageToPC(user, "Tag set to: " + args[0] + ".");
+                    }
+                });
+
+            _builder.Create("destination")
+                .Description("Sets the DESTINATION localstring to the given value on a target.")
+                .Permissions(AuthorizationLevel.DM, AuthorizationLevel.Admin)
+                .RequiresTarget()
+                .Validate((user, args) =>
+                {
+                    if (args.Length <= 0)
+                    {
+                        return "Missing arguments. Format should be: /destination <VALUE>. Example: /destination EventEntrance";
+                    }
+
+                    return string.Empty;
+                })
+                .Action((user, target, location, args) =>
+                {
+                    if (GetResRef(target) != "tele_obj")
+                    {
+                        SendMessageToPC(user, "Target is invalid. Please target a Teleport Object placeable.");
+                    }
+                    else
+                    {
+                        SetEventScript(target, EventScript.Placeable_OnUsed, "teleport");
+                        SetLocalString(target, "DESTINATION", args[0]);
+                        SendMessageToPC(user, "Applied teleport script to object. Destination tag set to " + args[0] + ".");
+                    }
+                }); 
         }
 
-        private void SetPortrait()
-        {
+            private void SetPortrait()
```

Comment:
Some unnecessary tabbing here - can you revert?

Reply:
oops, yeah